### ISSUE TITLE
Setting `services.sync.debug.ignoreCachedAuthCredentials` should reauth once per sync, not once per request

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/uniffi-rs#660]
We have a special setting `services.sync.debug.ignoreCachedAuthCredentials` that can be used when debugging auth-related changes in Firefox Sync. When set to `true`, it causes us to ignore cached auth credentials and instead request a fresh token every time we sync.

Here's the sequence of requests that I expected to trigger when doing a sync with this set:

* POST https://api.accounts.firefox.com/v1/certificate/sign
* GET https://token.servers.mozilla.com/1.0/sync/1.5
* GET https://sync-N.services.mozilla.com/1.5/12345/info/collections
* GET https://sync-N.services.mozilla.com/1.5/12345/info/configuration
* GET https://sync-N.services.mozilla.com/1.5/12345/storage/meta/global
* ...and so on for a full sync...

Here's the sequence of requests that actually happened:

* POST https://api.accounts.firefox.com/v1/certificate/sign
* GET https://token.servers.mozilla.com/1.0/sync/1.5
* GET https://sync-N.services.mozilla.com/1.5/12345/info/collections
* POST https://api.accounts.firefox.com/v1/certificate/sign
* GET https://token.servers.mozilla.com/1.0/sync/1.5
* GET https://sync-N.services.mozilla.com/1.5/12345/info/configuration
* POST https://api.accounts.firefox.com/v1/certificate/sign
* GET https://token.servers.mozilla.com/1.0/sync/1.5
* GET https://sync-N.services.mozilla.com/1.5/12345/storage/meta/global
* POST https://api.accounts.firefox.com/v1/certificate/sign
* GET https://token.servers.mozilla.com/1.0/sync/1.5
* ...and so on for a full sync...

In other words, it fetched a new BrowserID assertion and a new Tokenserver token for _every individual request_ made during a sync.

This seems surprising and counter-productive to me, making it potentially _harder_ to debug what's happening within a sync. I think a more useful behavior here would be to request a fresh tokenserver token at the stat of the sync and then use it for all requests during the sync. (But of course, it may be hard to actually implement that logic give the way the code is factored).

Thoughts, or counter-arguments that the current behaviour is actually desirable?

- - -
:beetle: Issue is synchronized with Bugzilla [Bug 1637156](https://bugzilla.mozilla.org/show_bug.cgi?id=1637156)
